src/video/video_atari.c

```diff
@@ -409,6 +521,18 @@ void Video_SetPalette(void *palette, int from, int length)
 			p += 3;
 		}
 		EsetPalette(from, length, rgb12);
+	} else if (s_machine_type == MCH_ST || s_machine_type == MCH_STE) {
+		uint8 red,green,blue;
+		for (i = from; i < from + length; i++)
+		{
+			red = *p++;
+			green = *p++;
+			blue = *p++;
+			s_palette4BitMap[i] = Palette_FindClosestColor(red, green, blue);
+		}
+		/* repaint only when a large amount of colors are changing, for fading and so on */
```

Comment:
only issue is that it won't update color palette for cycling colors (selected tile = color 255)
see https://github.com/OpenDUNE/OpenDUNE/blob/master/INTERNALS.txt#L40

also in map selection screen, colors 251,252,253,254 are cycling

Reply:
I agree!

Honestly, I am not sure what the best option is currently - I think it depends on the intent.
It could easily be changed to trigger the redraw at any color change. That would make it correct but at unplayable framerate on any machine the ST video mode is relevant for.

Ideally, I would detect the cycling on a lower level and redraw stuff effected by it. Possibly ignoring some cycling and do something else, depending on the performance target. But I feel this would start crossing the line where the engine/game is made to fit the target instead of the other way around.

As a compromise maybe it's possible to have the backend maintain a map of colors used by region and re-update effected regions. Not sure how feasible that is in terms of performance but maybe doable.

Reply:
I think you can make a static mapping to the color 255. and remap the other colors to only 15.
this will keep the blinking cursor without needing to redraw